### PR TITLE
Feat: Add account deletion button to Preferences

### PR DESCRIPTION
**Pull request by monicakh**
## What kind of change does this PR introduce?

Feature...

### CODE REVIEW

File: apps/studio/components/interfaces/Account/Preferences/AccountDeletion.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import React from 'react'
+import Panel from 'components/ui/Panel'
+import { Alert } from 'ui'
+import DeleteAccountButton from './DeleteAccountButton'
+import { useProfile } from 'lib/profile'
+
+const AccountDeletion = () => {
```
Comment (joshenlim) [inline apps/studio/components/interfaces/Account/Preferences/AccountDeletion.tsx:7]: `export const AccountDeletion` - we're avoiding export defaults as a code guideline

File: apps/studio/components/interfaces/Account/Preferences/index.ts [hunk 2]
```diff
@@ -3,3 +3,4 @@ export { default as AnalyticsSettings } from './AnalyticsSettings'
 export { default as ThemeSettingsOld } from './ThemeSettingsOld'
 export { default as ThemeSettings } from './ThemeSettings'
 export { default as ThemeSettingsWithPreferredTheme } from './ThemeSettingsWithPreferredTheme'
+export { default as AccountDeletion } from './AccountDeletion'
```
Comment (joshenlim) [inline apps/studio/components/interfaces/Account/Preferences/index.ts:6]: Wont need this then once we do export const above

File: apps/studio/components/interfaces/Account/Preferences/AccountDeletion.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  return (
+    <>
+      <Panel className="[&>div:first-child]:uppercase" title="Danger Zone">
+        <Panel.Content>
+          <Alert_Shadcn_ variant="destructive">
+            <CriticalIcon />
+            <AlertTitle_Shadcn_>Request for account deletion</AlertTitle_Shadcn_>
+            <AlertDescription_Shadcn_>
+              Deleting your account is permanent and cannot be undone. Your data will be deleted
+              within 30 days, except we may retain a limited set of data for longer where required
```
Comment (inian) [inline apps/studio/components/interfaces/Account/Preferences/AccountDeletion.tsx:19]:
```suggestion
              within 30 days, except we may retain some metadata and logs for longer where required
```

File: apps/studio/components/interfaces/Account/Preferences/AccountDeletion.tsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import Panel from 'components/ui/Panel'
+import { useProfile } from 'lib/profile'
+import { AlertDescription_Shadcn_, AlertTitle_Shadcn_, Alert_Shadcn_ } from 'ui'
+import { CriticalIcon } from 'ui-patterns/Icons/StatusIcons'
+import { DeleteAccountButton } from './DeleteAccountButton'
+
+export const AccountDeletion = () => {
+  const { profile } = useProfile()
+
+  return (
```
Comment (inian) [inline apps/studio/components/interfaces/Account/Preferences/AccountDeletion.tsx:10]: can we ensure the user has deleted all projects and orgs before we let them submit the ticket? (fine in a follow up PR). Less back and forth that way

Comment (monicakh) [reply apps/studio/components/interfaces/Account/Preferences/AccountDeletion.tsx:10]: Yes, that's a great suggestion and it's something I have already been looking into. I need to discuss it with the Frontend team first then follow up with another PR.

Comment (joshenlim) [reply apps/studio/components/interfaces/Account/Preferences/AccountDeletion.tsx:10]: no worries i'll look into it 🙂

**Review by joshenlim — APPROVED** [review]

**Pull request merged**